core/crypto/murmur/murmur.cpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto/murmur/murmur.hpp"
+
+#include <common/buffer.hpp>
+
+namespace fc::crypto::murmur {
+  constexpr uint64_t rotl64(uint64_t x, int8_t r) {
+    return (x << r) | (x >> (64 - r));
+  }
+
+  inline uint64_t fmix64(uint64_t k) {
+    k ^= k >> 33;
+    k *= 0xff51afd7ed558ccd;
+    k ^= k >> 33;
+    k *= 0xc4ceb9fe1a85ec53;
+    k ^= k >> 33;
+    return k;
+  }
+
+  std::vector<uint8_t> hash(gsl::span<const uint8_t> input) {
+    uint64_t h1 = 0;
+    uint64_t h2 = 0;
+
+    const uint64_t c1 = 0x87c37b91114253d5;
+    const uint64_t c2 = 0x4cf5ad432745937f;
+
+    //----------
+    // body
+
+    gsl::span<const uint64_t> blocks(
+        static_cast<const uint64_t *>(static_cast<const void *>(input.data())),
```

Comment:
Please rewrite without this awful reinterpret-like cast.
It is possible to avoid casts at all.